agent/app/agent.go

```diff
@@ -71,6 +71,8 @@ var (
 type agent interface {
 	// printVersion prints the Agent version string
 	printVersion() int
+	// printCapabilities prints the Agent's capabilities
+	printCapabilities() int
```

Comment:
I like the overall interface here. I'm a bit concerned that when future consumers of this interface see it, they will assume that capabilities are strictly bound to agent version.  We should make it clear that capabilities aren't only dependent on agent version, but also the environment and runtime version.